nixos/modules/programs/spacefm.nix

```diff
@@ -0,0 +1,41 @@
+# Global configuration for spacefm.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.spacefm;
+
+in
+{
+  ###### interface
+
+  options = {
+
+    programs.spacefm = {
+
+      settings = mkOption {
+        type = types.attrs;
+        default = {};
+        example = literalExample ''{
```

Comment:
I'll be sure to do just that if I ever end up doing a big system change. Thanks :)